core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/version/CodeSystemVersionCreateRequest.java

```diff
@@ -124,18 +124,58 @@ public Boolean execute(ServiceProvider context) {
 		} catch (NotFoundException e) {
 			// ignore
 		}
+
+		final Map<CodeSystemEntry, IVersioningManager> versioningManagersByCodeSystem = createVersioningManagers(context, codeSystem);
+		for (Entry<CodeSystemEntry, IVersioningManager> entry : versioningManagersByCodeSystem.entrySet()) {
+			createVersion(context, entry.getValue(), entry.getKey(), user);
+		}
 		
-		// check that the specified effective time
-		validateEffectiveTime(context, codeSystem);
+		return Boolean.TRUE;
+	}
+	
+	private Map<CodeSystemEntry, IVersioningManager> createVersioningManagers(final ServiceProvider context, final CodeSystemEntry codeSystem) {
+		final Map<CodeSystemEntry, IVersioningManager> versioningManagersByCodeSystem  = Maps.newHashMap();
+		
+		final IVersioningManager versioningManager = VersioningManagerBroker.INSTANCE.createVersioningManager(codeSystem.getTerminologyComponentId());
+		versioningManagersByCodeSystem.put(codeSystem, versioningManager);
+
+		if (versioningManager.needsOtherCodeSystemsToVersion()) {
```

Comment:
Not sure if this flag is really required, I would only use the `codeSystemDependenciesByRepositoryId()` method with a `forEach` if additional code systems need to be versioned along with the current one.